src/index.ts

```diff
@@ -21,21 +22,22 @@ export interface QRCode {
 
     bottomRightAlignmentPattern?: Point;
   };
+  colors?: QRColors;
 }
 
-function scan(matrix: BitMatrix): QRCode | null {
+function scan(matrix: BitMatrix, sourceData: Uint8ClampedArray, sourceWidth: number, scanOptions: Options): QRCode | null {
```

Comment:
Not sure if this makes sense or not, but the width and the height are already available on the binarized bitMatrix. Not sure we need to pass them in again.